docs/decisions/0008-course-key-table-for-cornerstone.rst

```diff
@@ -0,0 +1,40 @@
+Adding course key - uuid mapping for Cornerstone
+------------------------------------------------------
+
+Status
+======
+
+Draft
+
+Context
+=======
+
+At one point we were having transmitted courses to CSOD tenants fail with invalid chars contained within course metadata. 
+To combat this, we decided to encode the course keys. This has resulted in some of the courses failing validation with a 
+"course key too long" failures. There's a 50 char limit to course keys on the CSOD lms, so we need to limit the size of
+our encoded course keys within the content metadata.
+
+Simply truncating it would not allow for backwards compatibility or ensuring uniqueness. We also considered encoding
+and then compress, but while this would most likely work for every course key (the largest length found in prod was 54 chars),
+it can't properly guarantee the key to be under 50 chars, and we would have to worry with hashing the fact that 
```

Comment:
I would also mention the fact that we started an exploration of whether there are existing uuids we can use to map to course keys. We found that there are uuids in Discovery for course :
https://github.com/edx/course-discovery/blob/master/course_discovery/apps/course_metadata/models.py#L768
and 
courseruns: https://github.com/edx/course-discovery/blob/master/course_discovery/apps/course_metadata/models.py#L1277
 but we are still in the stages of finding out whether they are stable throughout their lifespan to be used. Turns out only courses with 'published' state in discovery have stable uuids. Because there are some unknowns there, this fix will allow us a more immediate remediation.